### PR TITLE
Adding Advanced Wireless Setup Kit

### PR DESCRIPTION
**Pull request by ReignOfFROZE**
This PR adds the Advanced Wireless Setup Kit, a tool aimed to ease the pain of binding many wireless connectors at once. It allows you to queue wireless connectors to be bound, and then bind them all in one go.

Has two modes
- Queuing Mode - Allows you to select wireless connectors to be bound
- Binding Mode - Binds the first wireless connector in the queue to the wireless connector that is selected
Sneak right click to change the mode of the tool.

Can be put in a crafting table to clear the NBT, which effectively will clear the queue as well to remove any left over/stale connectors

If anyone can come up with a better texture be my guest, I'm not great at sprite design.

Recipe PR here: https://github.com/GTNewHorizons/NewHorizonsCoreMod/pull/864

Video of it in action: 
https://github.com/GTNewHorizons/ae2stuff/assets/16232674/7748b4c6-96c3-448b-b0f8-f013d367be74

*misc addition is that I added an NBT clearing recipe for the normal wireless setup kit too, since sometimes it's hard to tell if that one is currently bound to a connector or not

### CODE REVIEW

**Comment by ReignOfFROZE** [comment]
Found a potential issue - marking as draft until I resolve it or confirm it's not an issue

**Comment by ReignOfFROZE** [comment]
Seems like the issue was a result of a bad build - unmarking as draft

File: src/main/scala/net/bdew/ae2stuff/items/AdvWirelessKit.scala [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+      yOff: Float,
+      zOff: Float
+  ): Boolean = {
+    import net.bdew.lib.helpers.ChatHelper._
+    val pos = BlockRef(x, y, z)
+    if (!pos.blockIs(world, BlockWireless)) return false
+    if (!world.isRemote) {
+      if (player.isSneaking) {
+        toggleMode(stack)
+        if (getMode(stack) == 0) {
```
Comment (Laiff) [inline src/main/scala/net/bdew/ae2stuff/items/AdvWirelessKit.scala:87]: Could you explicitly set different values for modes using two different constansts

File: src/main/scala/net/bdew/ae2stuff/misc/AdvItemLocationStore.scala [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      val tag = locList.getCompoundTagAt(i)
+      val pos = BlockRef.fromNBT(tag)
+      if (pos == loc) {
+        return false
+      }
+    }
+    locList.appendTag(NBT.from(loc.writeToNBT _))
+    tag.setTag("loc", locList)
+    tag.setInteger("dim", dimension)
+    print("Added location")
```
Comment (Laiff) [inline src/main/scala/net/bdew/ae2stuff/misc/AdvItemLocationStore.scala:36]: This log should check for debug enabled feature

File: src/main/scala/net/bdew/ae2stuff/misc/AdvItemLocationStore.scala [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+  def getLocations(stack: ItemStack) = {
+    if (!stack.hasTagCompound) stack.setTagCompound(new NBTTagCompound)
+    val tag = stack.getTagCompound
+    if (tag.hasKey("loc")) {
+      val locList = tag.getTagList("loc", 10)
+      locList
+    } else {
+      tag.setTag("loc", new NBTTagCompound)
+      tag.getTagList("loc", 10)
```
Comment (Laiff) [inline src/main/scala/net/bdew/ae2stuff/misc/AdvItemLocationStore.scala:49]: Avoid using magic constant for nbt type

File: src/main/scala/net/bdew/ae2stuff/items/AdvWirelessKit.scala [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+                          L("ae2stuff.wireless.tool.failed").setColor(Color.RED)
+                        )
+                      }
+                    } catch {
+                      case e: FailedConnection =>
+                        player.addChatComponentMessage(
+                          (L(
+                            "ae2stuff.wireless.tool.failed"
+                          ) & ": " & e.getMessage).setColor(Color.RED)
+                        )
```
Comment (Laiff) [inline src/main/scala/net/bdew/ae2stuff/items/AdvWirelessKit.scala:184]: Add log for full error here

**Review by Laiff — CHANGES_REQUESTED** [review]

File: src/main/scala/net/bdew/ae2stuff/items/AdvWirelessKit.scala (hunk 1 above)
Comment (ReignOfFROZE) [reply src/main/scala/net/bdew/ae2stuff/items/AdvWirelessKit.scala]: Resolved by d0fa5cc

File: src/main/scala/net/bdew/ae2stuff/misc/AdvItemLocationStore.scala (hunk 2 above)
Comment (ReignOfFROZE) [reply src/main/scala/net/bdew/ae2stuff/misc/AdvItemLocationStore.scala]: Resolved by d0fa5cc

File: src/main/scala/net/bdew/ae2stuff/misc/AdvItemLocationStore.scala (hunk 3 above)
Comment (ReignOfFROZE) [reply src/main/scala/net/bdew/ae2stuff/misc/AdvItemLocationStore.scala]: Resolved by d0fa5cc

File: src/main/scala/net/bdew/ae2stuff/items/AdvWirelessKit.scala (hunk 4 above)
Comment (ReignOfFROZE) [reply src/main/scala/net/bdew/ae2stuff/items/AdvWirelessKit.scala:187]: Resolved by d0fa5cc

File: src/main/scala/net/bdew/ae2stuff/misc/AdvItemLocationStore.scala (hunk 2 above)
Comment (ReignOfFROZE) [reply src/main/scala/net/bdew/ae2stuff/misc/AdvItemLocationStore.scala]: This was also unintentionally left in - was removed

**Review by Laiff — APPROVED** [review]

**Pull request merged**